evm/spec/tables/cpu.tex

```diff
@@ -1,4 +1,73 @@
+accordingly. The constraining of some complex instructions (e.g. Keccak hashing) is delegated to the other tables.
+This section will only briefly present the CPU and its columns. Details about the CPU logic will be provided later.
+
+\subsubsection{CPU flow}
+
+An execution run is made of three distinct parts:
+\begin{itemize}
+    \item \textbf{Bootstrapping:} The CPU starts by writing all the kernel code in memory, and hashing it. The hash is then compared to a public value shared with
+the verifier to ensure that the kernel code is correct.
+    \item \textbf{CPU cycles:} The bulk of the execution. In each row, the CPU reads the current code at the program counter (PC) address, and executes it. The current code can be the kernel code,
+or whichever code is being executed in the current context (transaction code or contract code). Executing an instruction consists in modifying the registers, possibly
+performing some memory operations, and updating the PC.
+\item \textbf{Padding:} At the end of the execution, we need to pad the length to the next power of two. Halting happens when the PC reaches the special halting label
+in the kernel. Constraints ensure that every subsequent row is a padding row and that you can't resume execution.
+\end{itemize}
+
+\subsubsection{CPU columns}
+
+\paragraph*{Registers:} \begin{itemize}
+    \item \texttt{is\_bootstrap\_kernel}: Boolean indicating whether this is a bootstrapping row or not. It must be 1 at the first row, then switch to 0 until the end.
+    \item \texttt{context}: Indicates which context we are in. 0 for the kernel, and a positive integer for every user context. Incremented by 1 at every call.
+    \item \texttt{code\_context}: Indicates in which context the code to execute resides. It's equal to \texttt{context} in user mode, but is always 0 in kernel mode.
+    \item \texttt{program\_counter}: The address of the instruction to be read and executed.
+    \item \texttt{stack\_len}: The current length of the stack.
+    \item \texttt{stack\_len\_bounds\_aux}: Helper column used to check that the stack doesn't overflow in user mode.
+    \item \texttt{is\_kernel\_mode}: Boolean indicating whether we are in kernel (i.e. privileged) mode. This means we are executing kernel code, and we have access to
+privileged instructions.
+    \item \texttt{gas}: The current amount of gas used in the current context. It is eventually checked to be below the current gas limit. Must fit in 32 bits.
+    \item \texttt{is\_keccak\_sponge}: Boolean indicating whether we are executing a Keccak hash. This happens whenever a \texttt{KECCAK\_GENERAL} instruction is executed, or at the last
+cycle of bootstrapping to hash the kernel code.
+    \item \texttt{clock}: Required to generate timestamps for memory operations to order them. Starts at 0 and is incremented by 1 at each row.
```

Comment:
These are not actual timestamps though. Perhaps rephrase this to explain that "clock" here is simply a monotonic counter used to enforce correct ordering of memory accesses?